src/test/java/org/folio/rest/impl/AcquisitionsUnitAssignmentsTest.java

```diff
@@ -0,0 +1,156 @@
+import java.util.UUID;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static org.folio.invoices.utils.ErrorCodes.MISMATCH_BETWEEN_ID_IN_PATH_AND_BODY;
+import static org.folio.rest.impl.InvoicesApiTest.BAD_QUERY;
+import static org.folio.rest.impl.MockServer.ERROR_X_OKAPI_TENANT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class AcquisitionsUnitAssignmentsTest extends ApiTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(AcquisitionsUnitAssignmentsTest.class);
+
+  private static final String ACQ_UNIT_ASSIGNMENTS_ENDPOINT = "/invoice/acquisitions-unit-assignments";
+
+  @Test
+  public void testGetAcqUnitAssignmentsNoQuery() {
+    logger.info("=== Test Get Acquisitions Unit Assignments - With empty query ===");
+    final AcquisitionsUnitAssignmentCollection units = verifySuccessGet(ACQ_UNIT_ASSIGNMENTS_ENDPOINT, AcquisitionsUnitAssignmentCollection.class);
+    assertThat(units.getAcquisitionsUnitAssignments(), hasSize(2));
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentsWithQuery() {
+    logger.info("=== Test GET Acquisitions Unit Assignments - search by query ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "?query=recordId==57257531-c6bc-4b6e-b91a-06bc0034b768";
+
+    final AcquisitionsUnitAssignmentCollection units = verifySuccessGet(url, AcquisitionsUnitAssignmentCollection.class);
+    assertThat(units.getAcquisitionsUnitAssignments(), hasSize(1));
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentsWithUnprocessableQuery() {
+    logger.info("=== Test GET Acquisitions Unit Assignments - unprocessable query ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "?query=" + BAD_QUERY;
+
+    verifyGet(url, APPLICATION_JSON, 400);
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentsSuccess() {
+    logger.info("=== Test GET Acquisitions Unit Assignment - success case ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/8ad580ee-d3f1-4271-bff4-c7c80ccb22e5";
+
+    final AcquisitionsUnitAssignment unit = verifySuccessGet(url, AcquisitionsUnitAssignment.class);
+    assertThat(unit, notNullValue());
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentNotFound() {
+    logger.info("=== Test GET Acquisitions Unit Assignment - not found ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/" + ID_DOES_NOT_EXIST;
+
+    verifyGet(url, APPLICATION_JSON, 404);
+  }
+
+  @Test
+  public void testPutAcqUnitAssignmentSuccess() {
+    logger.info("=== Test PUT Acquisitions Unit Assignment - success case ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/8ad510ee-d3f1-4271-bff4-c7c80ccb22e5";
```

Comment:
These UUID's can be stored as `static final` variables above

Reply:
Done